src/mavis/summary/summary.py

```diff
@@ -190,45 +192,92 @@ def group_by_distance(calls, distances):
     return grouped_calls, removed_calls
 
 
-def annotate_dgv(bpps, dgv_regions_by_reference_name, distance=0):
+def annotate_dgv(
+    bpps: List[BreakpointPair],
+    dgv_regions_by_reference_name: Dict[Tuple[str], List[BreakpointPair]],
+    input_cluster_radius,
+):
     """
-    given a list of bpps and a dgv reference, annotate the events that are within the set distance of both breakpoints
+    Given a list of breakpoint pairs (bpps) and bpps from another reference set), annotate the events that are within the set distance of both breakpoints
 
     Args:
         bpps (list) : the list of BreakpointPair objects
-        dgv_regions_by_reference_name (dict) : the dgv reference regions file loaded by load_masking_regions
-        distance (int) : the minimum distance required to match a dgv event with a breakpoint
+        dgv_regions_by_reference_name (dict) : tuple of (break1_chr,break2_chr) and its associated list of BreakpointPair/BioInterval objects specified by the MAVIS input file
+        cluster_radius (int) : Distance used in matching input SVs to reference SVs through clusterind, defined by summary.cluster_radius in the configuration file
     """
-    for chrom in dgv_regions_by_reference_name:
-        dgv_regions_by_reference_name[chrom] = sorted(
-            dgv_regions_by_reference_name[chrom], key=lambda x: x.start
-        )
 
-    lowest_resolution = max([len(b.break1) for b in bpps])  # only need start res
-
-    # only look at the bpps that dgv events could pair to, Intrachromosomal
-    for bpp in [
-        b for b in bpps if not b.interchromosomal and b.break1.chr in dgv_regions_by_reference_name
-    ]:
-        bpp.data['dgv'] = []
-        for dgv_region in dgv_regions_by_reference_name[bpp.break1.chr]:
-            dist = abs(Interval.dist(Interval(dgv_region.start), bpp.break1))
-            if dist > lowest_resolution + distance:
-                continue
-            elif (
-                dist > distance
-                or abs(Interval.dist(Interval(dgv_region.end), bpp.break2)) > distance
-            ):
-                continue
-            refname = dgv_region.reference_object
-            try:
-                refname = dgv_region.reference_object.name
-            except AttributeError:
-                pass
-            bpp.data['dgv'].append(
-                '{}({}:{}-{})'.format(dgv_region.name, refname, dgv_region.start, dgv_region.end)
+    if isinstance(list(dgv_regions_by_reference_name.values())[0][0], BreakpointPair):
+        bpps_copy = bpps.copy()
```

Comment:
This function is quite long now. Instead of this it would be cleaner to split it into two separate functions since they have no shared code

```
def annotate_dgv():
     if isinstance(list(dgv_regions_by_reference_name.values())[0][0], BreakpointPair):
         annotate_with_dgv_bpp(...)
     else:
         annoate_with_dgv_bed(...)
```

This also has the advantage of making a much smaller change diff overall

Reply:
Addressed in [960bd67](https://github.com/bcgsc/mavis/pull/336/commits/960bd67fdca774210a6cd5aac4c8c14bbe2a2a5f).